### PR TITLE
Fix 500 errors with ATOMIC_REQUESTS enabled (#85)

### LINKED ISSUE #85: Unavailable DB causes generic 500 if `ATOMIC_REQUESTS` is enabled
When [`ATOMIC_REQUESTS`](https://docs.djangoproject.com/en/1.9/ref/settings/#std:setting-DATABASE-ATOMIC_REQUESTS) is enabled any database error (e.g. server unreachable) causes the generic django 500 error page to be shown. 

This can be prevented by decorating the status view with [`@transaction.non_atomic_requests`](https://docs.djangoproject.com/en/1.9/topics/db/transactions/#django.db.transaction.non_atomic_requests)

### PR DESCRIPTION
**Pull request by ulope**
This PR fixes #85 

Along the way I also fixed the few Django 1.10 incompatibilities and updated the tox config to also locally test against Django 1.7 - 1.10. Also travis now uses tox to run the tests so there is no duplication. I hope you don't mind.

The fix itself is in the last commit (0690f8d). Without it the one new test (introduced in the commit before, f75e717) will fail.

### CODE REVIEW

**Comment by mwarkentin** [comment]
@ulope Thanks! I'll try to review this later today.

**Comment by mwarkentin** [comment]
@ulope - I've taken a quick look - looks like some really good testing updates in addition to your fix. I'll need to take a deeper look tomorrow.

Thanks again. :)

File: tests/test_decorators.py [hunk 1]
```diff
@@ -11,7 +11,10 @@
 
 import unittest
 
-from django.core.urlresolvers import reverse
+try:
+    from django.urls import reverse
```
Comment (mwarkentin) [inline tests/test_decorators.py]: Can you add comments here for which version of Django this and the fallback are for? Will make cleaning up when support for older versions is dropped easier.

File: tests/test_decorators.py [hunk 2]
```diff
@@ -33,35 +36,35 @@ def setUp(self):
 
     def test_200_ok_if_no_token_set(self):
         watchman_settings.WATCHMAN_TOKEN = None
-        response = self.client.get(reverse('watchman.views.status'))
+        response = self.client.get(reverse('status'))
```
Comment (mwarkentin) [inline tests/test_decorators.py]: Is this a Django compatibility thing?

**Comment by mwarkentin** [comment]
Just a couple of small questions, all in all looks really good.. thanks!

**Comment by mwarkentin** [comment]
@ulope Oh, you can also add yourself to the [authors file](https://github.com/mwarkentin/django-watchman/blob/master/AUTHORS.rst) if you want. 😄

**Comment by mwarkentin** [comment]
This PR should also fix #84

File: tests/test_decorators.py (hunk 1 above)
Comment (ulope) [inline tests/test_decorators.py]: The one in the try block is the "new" style introduced in 1.10

File: tests/test_decorators.py (hunk 2 above)
Comment (ulope) [inline tests/test_decorators.py]: Yes, reversing dotted module paths has been deprecated in 1.8 and was removed in 1.10.

**Comment by ulope** [comment]
Sorry, I didn't get to make your suggested changes in time, but I've responded to your comments above.

Anyway thanks for merging :)

**Comment by mwarkentin** [comment]
Thanks! Hoping to put out a release as soon as I can land #86.

**Comment by mwarkentin** [comment]
@ulope I've cut a 0.11rc here: https://pypi.python.org/pypi/django-watchman

Please give it a shot and let me know if you discover any issues. I'll cut the official release in a day or two assuming that I don't hear anything.

**Comment by mwarkentin** [comment]
@ulope I've released 0.11 including these changes: https://pypi.python.org/pypi/django-watchman

**Pull request merged**